### PR TITLE
Enhanced the RecoveryPlan-related struct.

### PR DESCRIPTION
**Pull request by prakash-nutanix**
Enhanced the RecoveryPlan-related struct to allow for specifying the category of volume groups to be included and to allow for specifying the cluster/AZ list.

### CODE REVIEW

File: v3/v3_structs.go [hunk 1]
```diff
@@ -2356,14 +2356,27 @@ type ProtectionRuleInput struct {
...
 // Parameters represents a object for resource of recovery plan
 type Parameters struct {
-	FloatingIPAssignmentList []*FloatingIPAssignmentList `json:"floating_ip_assignment_list,omitempty"`
-	NetworkMappingList       []*NetworkMappingList       `json:"network_mapping_list,omitempty"`
+	FloatingIPAssignmentList []*FloatingIPAssignmentList    `json:"floating_ip_assignment_list,omitempty"`
+	NetworkMappingList       []*NetworkMappingList          `json:"network_mapping_list,omitempty"`
+	AvailabilityZoneList     []*AvailabilityZoneInformation `json:"availability_zone_list,omitempty"`
+}
+
+type AvailabilityZoneInformation struct {
```
Comment (thunderboltsid) [inline v3/v3_structs.go:2377]: Please add a comment for all exported types.

Comment (prakash-nutanix) [reply v3/v3_structs.go]: @thunderboltsid I've addressed your comments. Please take a look.

**Review by thunderboltsid — APPROVED** [review]

**Review by thunderboltsid — DISMISSED** [review]

**Comment by thunderboltsid** [comment]
@prakash-nutanix can you also add a entry in CHANGELOG.md?

**Review by thunderboltsid — DISMISSED** [review]

File: v3/v3_service_test.go [hunk 2]
```diff
@@ -5206,7 +5206,37 @@ func TestOperations_CreateRecoveryPlan(t *testing.T) {
 			},
 			"spec": map[string]interface{}{
 				"resources": map[string]interface{}{
+        "volume_group_recovery_info_list": []interface{}{
```
Comment (deepakm-ntnx) [inline v3/v3_service_test.go:5209]: indentation seems to be off.

Comment (deepakm-ntnx) [reply v3/v3_service_test.go:5209]: pls perform make lint locally

**Comment by deepakm-ntnx** [comment]
need to rebase with latest master

**Review by deepakm-ntnx — APPROVED** [review]

**Review by thunderboltsid — APPROVED** [review]

**Pull request merged**